fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/effect/ServerMobEffectEvents.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.entity.event.v1.effect;
+
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.LivingEntity;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Events related to {@linkplain MobEffect status effects} in mobs.
+ *
+ * <p>Status effect events are useful when implementing generic behavior for
+ * many status effects and modifying the addition and removal behavior of
+ * existing status effects.
+ *
+ * <p>When only one class of {@linkplain MobEffect status effects}
+ * requires code to be run before the addition or removal of the effect,
+ * consider using {@link MobEffect#onEffectAdded(MobEffectInstance, LivingEntity)}
+ * or {@link MobEffect#onEffectRemoved(MobEffectInstance, LivingEntity)}.
```

Comment:
An alternative option would be to have a static method that returns the Event for a given MobEffect in a similar manner to how we did for game rules:

https://github.com/FabricMC/fabric/blob/dafb9388c88ff0a4ab7a6a8d88f42e18d298959c/fabric-game-rule-api-v1/src/main/java/net/fabricmc/fabric/api/gamerule/v1/GameRuleEvents.java#L32-L51

This is just an idea, it really depends if there is a common use case to be notified about all effects, if the main use case only cares about a specific effect maybe this would be a better way to go, and removes the need for the EventHandler interface.

Reply:
Honestly, I'm not sure about this. Personally, I'd like to see more opinion on this, but how I would use it would be, for example, an item that removes a list of effects or all effects. The way it is currently seems easier to use IMHO.